pkg/manager/member/tiflash_upgrader.go

```diff
@@ -101,6 +110,17 @@ func (u *tiflashUpgrader) Upgrade(tc *v1alpha1.TidbCluster, oldSet *apps.Statefu
 			if store.State != v1alpha1.TiKVStateUp {
 				return controller.RequeueErrorf("tidbcluster: [%s/%s]'s upgraded TiFlash pod: [%s], store state is not UP", ns, tcName, podName)
 			}
+			if ver, err := semver.NewVersion(tc.TiFlashVersion()); err == nil && ver.Compare(tiflashV512) >= 0 {
```

Comment:
If parse version failed, will skip this check.
Is it ok?

Reply:
For daily build with `nightly` or `latest` tag, we should also check this.

Reply:
Done